cmd/mistryd/metrics_recorder.go

```diff
@@ -0,0 +1,52 @@
+package main
+
+import (
+	"fmt"
+	"time"
+	"io/ioutil"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+type MetricsRecorder struct {
+	cfg *Config
+	buildMetrics *prometheus.GaugeVec
+}
+
+func NewMetricsRecorder(cfg *Config) *MetricsRecorder {
+	r := new(MetricsRecorder)
+	r.cfg = cfg
+
+	// The number of builds (per project)
+	r.buildMetrics = promauto.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "mistry_builds",
+			Help: "The number of builds",
+		},
+		[]string{"project"},
+	)
+
+	return r
+}
+
+func (r *MetricsRecorder) Start() {
+	for {
+		projects, _ := ioutil.ReadDir(r.cfg.ProjectsPath)
+
+		for _, project := range projects {
+			r.RecordBuildMetrics(project.Name())
+		}
+
+		time.Sleep(5 * time.Minute)
+	}
+}
+
+// Read the number of builds for the project by counting the folders under
+// its directories.
+func (r *MetricsRecorder) RecordBuildMetrics(project string) {
+	buildDir := fmt.Sprintf("%s/%s/ready", r.cfg.BuildPath, project)
```

Comment:
Is the number of builds going to count the contents of the `ready` folder or should we add the `pending` folder as well?

Reply:
So we have the "total number of builds" (an ever-increasing counter, counting all builds, pending+ready).

Now there are some extra dimensions:

1. A build can be either in-progress (pending) or completed (ready)
2. A completed build can be either successful or failed
3. A build can be either incremental or non-incremental